src/main/java/duke/util/Storage.java

```diff
@@ -0,0 +1,94 @@
+package duke.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.DukeException;
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * Storage class for storing and retrieving data from a file.
+ */
+public class Storage {
+
+    private final File dataFile;
+
+    /**
+     * Constructs a new Storage object with a given file path.
+     *
+     * @param path The path to the file to store data in.
+     */
+    public Storage(Path path) {
+        this.dataFile = new File(path.toUri());
+    }
+
+    /**
+     * Ensures that the {@code dataFile} is there and accessible.
+     */
+    private void ensureDataFile() {
```

Comment:
I like that the JavaDoc comment describes the comment well. But perhaps you could add 'Exists' to the end of the name for method name to be more descriptive.